File: src/main/java/net/adventuregame/story/Dialogue.java
package net.adventuregame.story;

import java.util.List;

public class Dialogue {
    private final String speaker;
    final List<String> lines;
    int currentLine = 0;

    public Dialogue(String speaker, List<String> lines) {
        this.speaker = speaker;
        this.lines = lines;
    }

    public boolean hasNext() {
        return currentLine < lines.size();
    }

    public String nextLine() {
        if (hasNext()) {
            return lines.get(currentLine++);
        }
        return null;
    }

    public void reset() {
        currentLine = 0;
    }

    public String getSpeaker() {
        return speaker;
    }
}

File: src/main/kotlin/net/adventuregame/story/Dialogue.kt
package net.adventuregame.story

class Dialogue(val speaker: String?, val lines: MutableList<String?>) {
    var currentLine: Int = 0

    fun hasNext(): Boolean {
        return currentLine < lines.size
    }

    fun nextLine(): String? {
        if (hasNext()) {
            return lines[currentLine++]
        }
        return null
    }

    fun reset() {
        currentLine = 0
    }
}

